src/app-layer-htp.c

```diff
@@ -1167,6 +1167,8 @@ static void FlagDetectStateNewFile(HtpTxUserData *tx, int dir)
             SCLogDebug("DETECT_ENGINE_STATE_FLAG_FILE_NEW set");
             tx->tx_data.de_state->dir_state[1].flags |= DETECT_ENGINE_STATE_FLAG_FILE_NEW;
         }
+        tx->tx_data.de_state->dir_state[DETECT_ENGINE_STATE_DIRECTION_BOTHDIR].flags |=
```

Comment:
so we update this always even for non-bidir sigs? Can we solve that differently?

Reply:
Well, this is not about signatures here : this is transaction data about files...
Indeed, we could avoid to do it if we do not have any bidir signatures...

Or am I missing something ?

Reply:
it's updated unconditionally, right?

Reply:
right... and so ?

Reply:
so also for non-bidir sigs

Reply:
And this line got removed anyways with your proposal in another point